api/v1beta1/openstackcluster_types.go

```diff
@@ -276,6 +276,8 @@ type OpenStackClusterStatus struct {
 	// and/or logged in the controller's output.
 	// +optional
 	FailureMessage *string `json:"failureMessage,omitempty"`
+
+	Conditions clusterv1.Conditions `json:"conditions,omitempty"`
```

Comment:
Once you remove this field from v1alpha* you'll see conversions tests failing. To please them you'll need to write restorers. The idea is that when you're adding new fields and not moving them around, there's no way a backward conversions won't be lossy. In order to make sure that v1beta1->v1alpha6->v1beta1 will return the same, some part of the CRD needs to be restored from information in the annotations. This is what the restorer will do. Please check out a few examples in the conversions code.

Reply:
If the test case covers the case you mentioned, then the code is okay. I cannot test this case in my local environment.